### PR TITLE
feat: APIify database access

### PR DESCRIPTION
**Pull request by thorwebdev**
Following the `cms-provider` approach for database access in preparation of adding a Supabase DB provider.

### CODE REVIEW

**Review by leerob — DISMISSED** [review]

**Comment by leerob** [comment]
CleanShot 2022-07-11 at 06 59 19@2x

Also, can you sign the CLA? 🙏

**Comment by thorwebdev** [comment]
Signed CLA sent and in the meantime I added in the supabase db provider.

**Comment by thorwebdev** [comment]
I'm running into and issue with the image rendering. 

This [answer on SO](https://stackoverflow.com/questions/66214552/tmp-chromium-error-while-loading-shared-libraries-libnss3-so-cannot-open-sha) suggest to use node v 14, but not sure if that's the correct resolution. Any ideas?

image

**Comment by thorwebdev** [comment]
Could it be related to the max serverless function size? But AFAICS it doesn't actually exceed it, no? 

image

**Comment by thorwebdev** [comment]
Okay, anyway, this issue seems unrelated to the changes in this PR, so I think we can go ahead and merge this in (it's running fine locally) and figure out the Vercel functions issue separately. I've opened a new issue for it here: https://github.com/vercel/virtual-event-starter-kit/issues/89

File: package.json [hunk 1]
```diff
@@ -35,7 +36,7 @@
     "ms": "2.1.3",
     "next": "^12.0.10",
     "nprogress": "0.2.0",
-    "puppeteer-core": "5.5.0",
+    "puppeteer-core": "10.1.0",
```
Comment (leerob) [inline package.json:39]: Let’s revert this change

Comment (leerob) [inline package.json:39]:
```suggestion
    "puppeteer-core": "5.5.0",
```

**Comment by leerob** [comment]
Might need to update the lock file

**Review by leerob — APPROVED** [review]

**Comment by Deep-Codes** [comment]
Register is failing 🤔 @thorwebdev 

Screenshot 2022-09-02 at 6 34 12 PM

**Comment by thorwebdev** [comment]
@Deep-Codes do you have any details on the error message of the API route?
